prebuilt_swift_static_framework/prebuilt_swift_static_framework.bzl

```diff
@@ -21,40 +27,73 @@ def _zip_swift_arg(module_name, swift_identifier, input_file):
         file_path = input_file.path,
     )
 
+def _zip_generated_objc_hdr_arg(module_name, generated_objc_hdr_file):
+    # Rename the generated ObjC header to `{module_name}-Swift.h`.
+    return "{module_name}.framework/Headers/{module_name}-Swift.h={file_path}".format(
+        module_name = module_name,
+        file_path = generated_objc_hdr_file.path,
+    )
+
 def _prebuilt_swift_static_framework_impl(ctx):
-    module_name = ctx.attr.framework_name
+    module_name = ctx.attr.module_name
     fat_file = ctx.outputs.fat_file
 
-    input_archives = []
-    input_modules_docs = []
-    zip_args = [_zip_binary_arg(module_name, fat_file)]
+    input_libraries = []
+    input_swift_info = []
+    generated_objc_hdr_file = None
+
+    # The Swift static framework has only one `swift_library` dependency target.
+    platform_to_target_list = ctx.split_attr.archive.items()
+    swift_library_target = platform_to_target_list[0][1]
+
+    # Get the generated Objective-C header file for the `swift_library`.
+    objc_provider = swift_library_target[apple_common.Objc]
```

Comment:
Are `swift_library`s guaranteed to have a `apple_common.Objc` provider? Or should this check first?

Reply:
Good point, only if the toolchain supports ObjC interop, so should add a check to future-proof.

https://github.com/bazelbuild/rules_swift/blob/8a7fdf43a8a0c24b133ad8821d040642a4cd5a66/swift/internal/swift_library.bzl#L149